lib/jekyll-admin/server/page.rb

```diff
@@ -12,6 +12,11 @@ class Server < Sinatra::Base
       end
 
       put "/*?/?:path.:ext" do
+        unless html_content?
+          status 422
+          content_type :json
+          return json "error_message" => "Invalid file extension for pages"
```

Comment:
I'd make the message more descriptive, e.g., `Files must have an extension of X, Y, or Z`.